src/server/gun/gun-middleware.js

```diff
@@ -5,12 +5,24 @@ import SEA from 'gun/sea'
 import 'gun/lib/load'
 import { memoize } from 'lodash'
 // import les from 'gun/lib/les'
-import { type StorageAPI } from '../../imports/types'
+import { wrapAsync } from '../utils/helpers'
+import { LoggedUser, type StorageAPI } from '../../imports/types'
 import conf from '../server.config'
 import logger from '../../imports/logger'
-
+import { sha3 } from 'web3-utils'
 const log = logger.child({ from: 'GunDB-Middleware' })
 
+Gun.chain.putAck = function(data, cb) {
+  var gun = this,
+    callback =
+      cb ||
+      function(ctx) {
+        return ctx
+      }
+  let promise = new Promise((res, rej) => gun.put(data, ack => (ack.err ? rej(ack) : res(ack))))
```

Comment:
why is using 'once' required?

Reply:
@sirpy for GC could collect unused promise
otherwise further .put() callback calls will hold reject/resolve references, and (as a consequence) the promise object